### DISCUSSION TITLE
LLDP neighbors list for Huawei switches is empty

### DISCUSSION THREAD

**Discussion by fesenkokd** [netbox-community/netbox#11534] [Q&A]
Hello,
please help to tshoot Netbox empty lldp neighbor tab for Huawei switch. I'm using napalm-ce module.
When I exec GET to Netbox API I'm receiving switch LLDP neighbors(/napalm/?method=get_lldp_neighbors).
`{
    "get_lldp_neighbors": {
        "100GE1/0/1": [
            {
                "port": "100GE1/0/1",
                "hostname": "tas-ldc-leaf-sw-01"
            }
        ],
        "100GE1/0/2": [
            {
                "port": "100GE1/0/2",
                "hostname": "tas-ldc-leaf-sw-01"
            }
        ]
    }
}`
But in netbox device tab is empty:
![image](https://user-images.githubusercontent.com/57910046/213386639-bba76d22-92b5-453f-9c5c-a8787f93380a.png)
Interface names are matching. What I'm doing wrong?
Also, what method Netbox uses for this tab - get_lldp_neighbors_detail or get_lldp_neighbors ?

**Comment by candlerb** [comment]
Most likely causes are browser to API authentication, or a problem with interface naming.

If your Netbox interface is called "100GE1/0/1", but the response from the Huawei is different, then it can't find a matching row in the table to update.  LLDP results for unknown interfaces are unfortunately thrown away.  However your example seems to show it's correct.

Try opening your browser's developer tools, and seeing what happens when you switch to the LLDP Neighbors tab.
* are there errors logged on the console?
* is there network activity to the following URLs?  In my browser I see:
    * /dcim/devices/NN/lldp-neighbors/
    * ...
    * /api/dcim/devices/NN/napalm?method=get_lldp_neighbors_detail
* For the latter URL, what's the status code and response body shown in the browser?

> Also, what method Netbox uses for this tab - get_lldp_neighbors_detail or get_lldp_neighbors ?

The second - see above. The first URL displays the HTML with the database-derived table of interfaces.  The second is called from Javascript to update the table with LLDP neighbor information.

**Reply by fesenkokd** [reply]
Hello,
Brian, thank you for reply, I checked in browser dev mode - there is status code - 200.
requested url - dcim/devices/61/lldp-neighbors/  
![image](https://user-images.githubusercontent.com/57910046/213635693-dd8e9f1f-1c72-47b3-8fd0-ef1bcb01785b.png)
![image](https://user-images.githubusercontent.com/57910046/213635945-8f967e09-80c9-43e7-9d41-a4c43c2d2c92.png)
And error in response - {"get_lldp_neighbors_detail":{"error":"Method get_lldp_neighbors_detail not implemented for NAPALM driver <class 'napalm_ce.ce.CEDriver'>"}}
![image](https://user-images.githubusercontent.com/57910046/213637993-fbe48b96-612b-40f5-9e37-f46c2e3073ad.png)

I also repeated method **get_lldp_neighbors** from https://github.com/napalm-automation-community/napalm-ce/blob/master/napalm_ce/ce.py 
Connected with netmiko from my pc and output is:
`{'100GE1/0/1': [{'port': '100GE1/0/1', 'hostname': 'tas-ldc-leaf-sw-02'}], '100GE1/0/2': [{'port': '100GE1/0/2', 'hostname': 'tas-ldc-leaf-sw-02'}]}`
There is no blank symbols in interface naming.
For Cisco Device all works fine:
![image](https://user-images.githubusercontent.com/57910046/213637691-388c78e5-2df3-4565-a2aa-ddc62afefa55.png)

**Reply by candlerb** [reply]
> And error in response - {"get_lldp_neighbors_detail":{"error":"Method get_lldp_neighbors_detail not implemented for NAPALM driver <class 'napalm_ce.ce.CEDriver'>"}}

That'll be it then. You'll need to take this up with whoever maintains that Huawei napalm driver.
